### ISSUE TITLE
Some keyboard layouts crash Phoenix when binding keys

### ISSUE THREAD

**Issue by mumoshu** [kasper/phoenix#196]
- Version: 2.5
- macOS: Sierra

Hi, thanks for maintaining the great project 👍 
I've recently tried using phoenix and encountered an issue and any `Key.on` call result in sudden-death of phoenix process with console log like seen in https://github.com/kasper/phoenix/issues/34.

## Possible cause

After digging a bit, the improvement made in https://github.com/kasper/phoenix/commit/cc5dafc2fc4914bbc3f08c17bbdce6ecabee7e8a seems to be insufficient in my case.

Concretely speaking,  `TISCopyCurrentKeyboard"Layout"InputSource` should be used instead of `TISCopyCurrent"ASCIICapable"KeyboardInputSource` and `TISCopyCurrentKeyboardInput`.
Otherwise the layout data becomes null and `CFDataGetBytePtr (layoutData)` results in a bad access.

## Known work-around

Switch your input mode to the one provided by macOS - "ABC"(Alphabet input mode, not sure how it looks like in users in locales other than jpn though):

![image](https://user-images.githubusercontent.com/22009/33011203-97b329d6-ce20-11e7-8ed0-b5192d894dda.png)

And then run phoenix as usual. The issue will go away.

## Steps to reproduce

If you've like to reproduce the issue, you can install [Google IME for Japanese](https://www.google.co.jp/ime/) and switch to either:

- the Hiragana-input mode or
- the ABC(Alphabet)-input mode

and then run phoenix with single configuration containing just `Key.on(...)` in it. It will immediately quit without any noticeable error message, except a dump seen in #34.

You can check different values for layoutData and keyboardLayout obtained by TISCopyCurrentKeyboardInputSource variants using [the checker app in my gist](https://gist.github.com/mumoshu/ff53cf33432843ed4e88edd42d78b436).

With my ASCII keyboard w/ Google IME(for Japanese input) switched to the Hiragana-input mode rather than alphabet-input mode, what I get with the checker app is:

```
$ make && ./InputSourceChecker
make: Nothing to be done for `all'.
From TISCopyCurrentKeyboardLayoutInputSource: InputSourceID=com.apple.keylayout.ABC LocalizedName=ABC layoutData=<not null> keyboardLayout=<not null>
From TISCopyCurrentASCIICapableKeyboardInputSource: InputSourceID=com.google.inputmethod.Japanese.Roman LocalizedName=Alphanumeric (Google) layoutData=<null> keyboardLayout=<null>
From TISCopyCurrentKeyboardInputSource: InputSourceID=com.google.inputmethod.Japanese.Roman LocalizedName=Alphanumeric (Google) layoutData=<null> keyboardLayout=<null>
```

**Comment by kasper** [comment]
Hi! Thanks for the report, I’ll try to figure it out. If you have time, feel free to make a PR with the fix.

**Comment by kasper** [comment]
Possible related to #166.

**Comment by kasper** [comment]
A bit embarrassing how long it took to fix this, but hopefully this should now be properly fixed in 2.6.4.

**Comment by kasper** [comment]
@mumoshu Thanks for the detailed explanation and sample code!
